src/main/java/io/cdap/plugin/gcp/bigtable/sink/RecordToHBaseMutationTransformer.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.gcp.bigtable.sink;
+
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.format.UnexpectedFormatException;
+import io.cdap.cdap.api.data.schema.Schema;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.util.Bytes;
+
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Transforms Google Cloud Bigtable {@link Result} to {@link StructuredRecord}.
+ */
+public class RecordToHBaseMutationTransformer {
+
+  private final String keyAlias;
+  private final String columnFamily;
+
+  public RecordToHBaseMutationTransformer(String keyAlias, String columnFamily) {
+    this.keyAlias = keyAlias;
+    this.columnFamily = columnFamily;
+  }
+
+  public Mutation transform(StructuredRecord record) {
+    List<Schema.Field> fields = Objects.requireNonNull(record.getSchema().getFields(), "Schema fields cannot be empty");
+    String key = Objects.requireNonNull(record.get(keyAlias), "Key cannot be null");
```

Comment:
We should not be assuming the key is a string. The plugin should allow any of the supported types as a key.

Reply:
Include the field name in the error message:

"Key field '%s' contained a null value. Keys cannot be null."

Reply:
Changed logic to allow key to be of any supported type.